lib/internal/webstreams/adapters.js

```diff
@@ -615,9 +637,11 @@ function newReadableWritablePairFromDuplex(duplex) {
     throw new ERR_INVALID_ARG_TYPE('duplex', 'stream.Duplex', duplex);
   }
 
+  validateObject(options, 'options');
+
   if (isDestroyed(duplex)) {
     const writable = new WritableStream();
-    const readable = new ReadableStream();
+    const readable = new ReadableStream(options);
```

Comment:
Oh, something else: because we're passing the *entire* `options` object here, this might cause unintended results if you pass extra properties that are recognized by [the `ReadableStream` constructor](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/ReadableStream#underlyingsource):
```javascript
duplex.toWeb({
  type: "bytes",
  start(controller) {
    controller.enqueue("weird")
  }
})
```
We should probably only pass the `type` here.
```suggestion
    const readable = new ReadableStream({ type: options.type });
```

Reply:
Sorry, I typo'd. I meant:
```javascript
Duplex.toWeb(duplex, {
  type: "bytes",
  start(controller) {
    controller.enqueue("weird")
  }
})
```
But sure, you could argue that this is undefined behavior and you shouldn't pass extra properties in the `options` bag in the first place. How strict should we be?

Reply:
Just to be sure I committed your sugesstion.